packages/main/src/Menu.js

```diff
@@ -495,6 +571,40 @@ class Menu extends UI5Element {
 			this._prepareSubMenuDesktopTablet(item, opener, actionId);
 		}
 	}
+
+	_beforePopoverOpen(event) {
+		const prevented = !this.fireEvent("before-open", {}, true, false);
+
+		if (prevented) {
+			this.open = false;
```

Comment:
I would prefer having opened as boolean variable. Open is a verb and i would expect a function.

Reply:
it is already implemented as "open" in Popup, and Popover, ResponsivePopover and Dialog inherit it also.

Reply:
You can find open/closed on shops, markets - it's not only the verb/acroon"open", but refers to the state as well